src/Client.php

```diff
@@ -381,6 +426,43 @@ private function extractSpaceId(array $data)
         return '[blank]';
     }
 
+    /**
+     * Checks a data structure and extracts the environment ID, if present.
+     *
+     * @param array $data
+     *
+     * @return string|null
+     */
+    public function extractEnvironmentId(array $data)
+    {
+        // Space resource
+        if (isset($data['sys']['type']) && $data['sys']['type'] === 'Space') {
+            return $this->environmentId;
+        }
+
+        // Environment resource
+        if (isset($data['sys']['type']) && $data['sys']['type'] === 'Environment') {
+            return $data['sys']['id'];
+        }
+
+        // Resource linked to a environment
+        if (isset($data['sys']['environment'])) {
+            return $data['sys']['environment']['sys']['id'];
+        }
+
+        // Array resource with at least an element
+        if (isset($data['items'][0]['sys']['environment'])) {
+            return $data['items'][0]['sys']['environment']['sys']['id'];
+        }
+
+        // Empty array resource
+        if (isset($data['items']) && !$data['items']) {
+            return $this->environmentId;
+        }
+
+        return '[blank]';
```

Comment:
Why `'[blank]'` and not `'master'`?

Reply:
I'm debating this.

Using `master` would be helpful for backwards compatibility, because a pre-environments JSON string would be correctly recognized, but on the other hand if you reach this point you have done something weird.

I think I'll change it to `master` for now, but I'll probably deprecate this sometimes in the future 👍